dlt/sources/rest_api/__init__.py

```diff
@@ -288,6 +294,7 @@ def process(
             def paginate_resource(
                 method: HTTPMethodBasic,
                 path: str,
+                headers: Dict[str, Any],
```

Comment:
```suggestion
                headers: Optional[Dict[str, Any]],
```

should be optional as we use `dict.get()`